compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -509,6 +515,49 @@ encodeModuleRule =
         let (hash, bs) = SS.encodeModule lfVersion m
         return ([], Just (mconcat $ hash : map fst encodedDeps, bs))
 
+-- hlint
+
+hlintSettings :: FilePath -> IO ([Classify], Hint)
+hlintSettings hlintDataDir = do
+  -- `findSettings` ends up calling `readFilesConfig` which in turn
+  -- calls `readFileConfigYaml` which finally calls `decodeFileEither`
+  -- from the `yaml` library.  Annoyingly that function catches async
+  -- exceptions and in particular, it ends up catching `ThreadKilled`.
+  -- So, we have to mask to stop it from doing that.
+  (_, classify, hints) <- mask $ \unmask ->
+    findSettings (unmask . readSettingsFile (Just hlintDataDir)) Nothing
+  return (classify, hints)
+
+getHlintDiagnosticsRule :: Options -> Rules ()
+getHlintDiagnosticsRule opts =
+    define $ \GetHlintDiagnostics file -> do
+        pm <- use_ GetParsedModule file
+        let anns = pm_annotations pm
+        let modu = pm_parsed_source pm
+        case optHlintDataDir opts of
+          Just dir -> do
+            (classify, hint) <- liftIO $ hlintSettings dir
+            let ideas = applyHints classify hint [createModuleEx anns modu]
+            return ([toDiagnostic file i | i <- ideas, ideaSeverity i /= Ignore], Just ())
+          Nothing -> do
+            logger <- actionLogger
+            liftIO $ logError logger $ T.pack $
```

Comment:
This should just call `fail` and not emit a log message.

Reply:
Pushed caching logic just now. No longer logs, just fails.